include/internal/to_hex.h

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2024 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#ifndef OSSL_INTERNAL_TO_HEX_H
+# define OSSL_INTERNAL_TO_HEX_H
+# pragma once
+
+static ossl_inline size_t to_hex(char *buf, uint8_t n, const char hexdig[17])
```

Comment:
This function ends up in the namespace of the including module as much as the next one, so really ought to have the `ossl_` prefix on it too. That then clashes with `ossl_to_hex` in `cryptlib.h`, so this should probably be `ossl_to_hex_internal()`, since generally the caller shouldn't need to tell us what alphabet they want to use, only if they want upper- or lowercase output.

Reply:
Still this is a static function in an internal header, so I do not think the ossl_ prefix is necessary. If there are ever any conflicts with the file where the internal header is included, they can be resolved in that change.

Reply:
Happy for this to be handled later. @t8m I think this is the sort of thing that is likely to get included by several different modules, as it's so generic, so it seems to me to make sense to avoid those issues from the beginning.